src/game/process/cmd/attack.mjs

```diff
@@ -46,6 +46,11 @@ export function attack(state, action, timestamp, horde) {
     prevPos = attackPos;
   }
 
+  if (opponent && state.teamsConfig.on && opponent.team.id == player.team.id) {
```

Comment:
I believe an "opponent" that is from the same team should be considered in the exact same way an an obstacle - so it should block the attack, unless the attacker is the hacker_beaver (who has an ability to shoot "through" the obstacles)